index.js

```diff
@@ -71,7 +71,15 @@ function redisStore(args) {
       }
 
       if (opts.parse) {
-        result = JSON.parse(result);
+
+        try {
+          // allow undefined only if allowed by isCacheableValue
+          if(! ( (result === undefined || result === "undefined") && typeof args.isCacheableValue === 'function' && args.isCacheableValue(result))) {
```

Comment:
Not sure what part to skip, the entire if section ?
Result can come as undefined and "undefined" and causes failure on JSON.parse
In case the value is not cacheable yes this will fail (if will allow to parse undefined), but this is expected and since it is in a try catch block you will get that error in your callback.
isCachebaleValue by default does not allow undefined.
There is a case here (referring to your node_redis link) where the key exists and value is 'undefined'.